src/Processors/Pipe.h

```diff
@@ -101,7 +101,7 @@ class Pipe
     void setQuota(const std::shared_ptr<const EnabledQuota> & quota);
 
     /// Do not allow to change the table while the processors of pipe are alive.
-    void addTableLock(const TableLockHolder & lock) { holder.table_locks.push_back(lock); }
+    void addTableLock(TableLockHolder lock) { holder.table_locks.emplace_back(std::move(lock)); }
```

Comment:
If return value of `emplace_back(&&)` is not used at the call site it is equivalent to `push_back(&&)`.

Reply:
Sure